repo.go

```diff
@@ -22,22 +22,23 @@ type Interface interface {
 	// GetBundle returns the bundle referenced by curl.
 	GetBundle(curl *charm.URL) (charm.Bundle, error)
 
-	// Latest returns the latest revision of the charms referenced by curls,
-	// regardless of the revision set on each curl.
-	Latest(curls ...*charm.URL) ([]CharmRevision, error)
-
-	// Resolve resolves the series and revision of the given entity
-	// reference. If the series is not specified, it may be resolved
-	// by the charm store or rejected. After the series is resolved,
-	// if the revision is not specified, it will be resolved to the latest
-	// available revision for that series.
-	Resolve(ref *charm.Reference) (*charm.URL, error)
+	// Resolve resolves revision of the given entity reference and
+	// in addition, returns which series the entity may declare it
+	// supports. If the series is not specified, the caller may pick
+	// a series from those supported by the entity. If a series is
+	// specified and it is not supported, an error is returned.
+	// If the revision is not specified, it will be resolved to the
+	// latest available revision for that entity, possibly accounting
+	// for series if it is specified.
+	// The second return value holds the list of possible series
+	// supported by the entity, with the preferred series first.
```

Comment:
// It may be empty if the charm does not declare any Series in its metadata.

?

Or should that be:

// It will be empty if the charm does not declare any Series in its metadata.

I'm thinking that the latter might be better as it actually provides an indicator
of whether the charm may support multiple series of not.

I'm actually wondering about something like:

    // Resolve resolves the given reference to a canonical form which refers
    // unambiguously to a specific revision of an entity. If the entity
    // is a charm that may support more than one series, canonRef.Series will
    // be empty and supportedSeries will hold the list of series supported by
    // the charm with the preferred series first.
    // If ref holds a series, then Resolve will always ensure that the returned
    // entity supports that series.
    Resolve(ref *charm.Reference) (canonRef *charm.Reference, supportedSeries  []string, err error)

AFAICS there is actually no particular need to change this interface now, as the charm store does not yet support multi-series charms, but I'm OK to do so.